docs/Segments.md

```diff
@@ -312,7 +322,118 @@ Used by certain compilers (like GCC) to store the Exception Handler Frame, used
 
 This frame contains more metadata used by exceptions at runtime.
 
-## PS2 exclusive segments
+## `linker_offset`
+
+This segment adds a symbol into the linker script at its relative section position.
+
+A segment named "john" with type `linker_offset` will cause a generated symbol with the name `john_OFFSET` to be placed into the linker script.
+This can be useful for naming and referencing certain address locations from source code.
+
+# Platform-specific segments
+
+## N64
+
+### Images
+
+**Description:**
+
+**splat** supports most of the [N64 image formats](https://n64squid.com/homebrew/n64-sdk/textures/image-formats/):
+
+- `i`, i.e. `i4` and `i8`
+- `ia`, i.e. `ia4`, `ia8`, and `ia16`
+- `ci`, i.e. `ci4` and `ci8`
+- `rgb`, i.e. `rgba32` and `rgba16`
+
+These segments will parse the image data and dump out a `png` file.
+
+**Note:** Using the dictionary syntax allows for richer configuration.
+
+**Example:**
+
+```yaml
+# as list
+- [0xABC, i4, filename, width, height]
+# as a dictionary
+- name: filename
+  type: i4
+  start: 0xABC
+  width: 64
+  height: 64
+  flip_x: yes
+  flip_y: no
+```
+
+`ci` (paletted) segments have a `palettes: []` setting that represents the list of palettes that should be linked to the `ci`. For each linked palette, an image will be exported. The implicit value of `palettes` is a one-element list containing the name of the raster, which means palettes and rasters with the same name will automatically be linked.
+
+Palette segments can specify a `global_id`, which can be referred to from a `ci`'s `palettes` list. The `global_id` space is searched first, and this allows cross-segment links between palettes and rasters.
+
+We recommend using [pigment64](https://github.com/decompals/pigment64) to convert extracted images back into original formats.
+
+### `gfx`
+
+`gfx` can be used to extract static f3dex ["display lists"](https://hackmd.io/@Roman971/Hk01jRxRr#Static-Data) into a .gfx.inc.c file, which is meant to be `#include`d from a source c file.
+
+These segments support an optional `data_only` attribute, which is False by default. If enabled, the extracted file will contain only the data rather than the enclosing symbol definition.
+
+Example output with `data_only` off (default):
+
+```
+Gfx displayList[] = {
+    gsDPPipeSync(),
+    gsDPSetPrimColor(0, 0, 0x80, 0x80, 0x80, 0x80),
+    gsDPSetEnvColor(0x80, 0x80, 0x80, 0x80),
+    gsSPEndDisplayList(),
+};
+```
+
+to be used in a source c file like
+```
+#include "example.gfx.inc.c"
+```
+
+Example output with `data_only` on:
+```
+gsDPPipeSync(),
+gsDPSetPrimColor(0, 0, 0x80, 0x80, 0x80, 0x80),
+gsDPSetEnvColor(0x80, 0x80, 0x80, 0x80),
+gsSPEndDisplayList(),
+```
+
+to be used in a source c file like
+```
+Gfx displayList[] = { #include "example.gfx.inc.c" };
```

Comment:
I think the include can't be on the middle of the line.
```suggestion
```c
Gfx displayList[] = {
#include "example.gfx.inc.c"
};
```

I may be misremembering tho